### PR TITLE
Fix a bug in validating node existence.

### LINKED ISSUE #57483: An unexpected error can cause nodes in Azure to go NotReady
There is a bug where if an error returned when looking up a node is not an instance of `autorest.DetailedError` that the azure client will mistakenly return `not-found, nil` instead of the error.

This can cause a variety of problems, including nodes going into the `NotReady` state.

### PR DESCRIPTION
**Pull request by brendandburns**
**What this PR does / why we need it**:
Fixes an bug where if an error was returned that was not an `autorest.DetailedError` we would return `"not found", nil` which would result in Nodes going `NotReady`

**Which issue(s) this PR fixes** 
Fixes #57483

**Release note**:
```release-note
Fixes a bug where if an error was returned that was not an `autorest.DetailedError` we would return `"not found", nil` which caused nodes to go to `NotReady` state.
```

@feiskyer @khendiak

### CODE REVIEW

File: pkg/cloudprovider/providers/azure/azure_wrap_test.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2016 The Kubernetes Authors.
```
Comment (itowlson) [inline pkg/cloudprovider/providers/azure/azure_wrap_test.go]: ridiculous pedantry: should this be 2017?

File: pkg/cloudprovider/providers/azure/azure_wrap_test.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+limitations under the License.
+*/
+
+package azure
+
+import (
+	"fmt"
+        "net/http"
+        "reflect"
+	"testing"
```
Comment (itowlson) [inline pkg/cloudprovider/providers/azure/azure_wrap_test.go]: It looks like the alignment in the import section is inconsistent, or is that the GitHub UI?

**Review by itowlson — APPROVED** [review]

**Comment by brendandburns** [comment]
Once this merges...

@jdumars can we see about getting this cherry picked into 1.8.x and 1.9.x (and possibly 1.7.x?)

Thanks

**Comment by jdumars** [comment]
On it @brendandburns

**Comment by jdumars** [comment]
@brendandburns I ran into some merge conflicts with azure.go on the the 1.8 cherry pick, and am not sure I can do it safely.

**Comment by brendandburns** [comment]
Ok I'll look and send a PR direct on the 1.8 branch...

--brendan

________________________________

**Pull request merged**